crypto/ex_data.c

```diff
@@ -287,8 +287,6 @@ int CRYPTO_dup_ex_data(int class_index, CRYPTO_EX_DATA *to,
         CRYPTOerr(CRYPTO_F_CRYPTO_DUP_EX_DATA, ERR_R_MALLOC_FAILURE);
         return 0;
     }
-    if (!CRYPTO_set_ex_data(to, mx - 1, NULL))
-        goto err;
```

Comment:
but if this function is used on a previously initialized objects it could have more elements.
Then you have a mix of old an new object slots.
We could do CRYPTO_set_ex_data(to, mx - 1, CRYPTO_get_ex_data(to, mx -1))
but maybe it would be better to assert that this does not happen.